spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/WellKnownAuthParameters.java

```diff
@@ -24,48 +24,70 @@
  *
  * @author Alexander Preuß
  */
-public final class WellKnownAuthParameters {
+enum WellKnownAuthParameters {
+
+	TENANT_DOMAIN("tenantDomain"),
+
+	TENANT_SERVICE("tenantService"),
+
+	PROVIDER_DOMAIN("providerDomain"),
+
+	PRIVATE_KEY("privateKey"),
+
+	PRIVATE_KEY_PATH("privateKeyPath"),
+
+	KEY_ID("keyId"),
+
+	AUTO_PREFETCH_ENABLED("autoPrefetchEnabled"),
+
+	ATHENZ_CONF_PATH("athenzConfPath"),
+
+	PRINCIPAL_HEADER("principalHeader"),
+
+	ROLE_HEADER("roleHeader"),
+
+	ZTS_URL("ztsUrl"),
+
+	USER_ID("userId"),
+
+	PASSWORD("password"),
+
+	KEY_STORE_TYPE("keyStoreType"),
+
+	KEY_STORE_PATH("keyStorePath"),
+
+	KEY_STORE_PASSWORD("keyStorePassword"),
+
+	TYPE("type"),
+
+	ISSUER_URL("issuerUrl"),
+
+	AUDIENCE("audience"),
+
+	SCOPE("scope"),
+
+	SASL_JAAS_CLIENT_SECTION_NAME("saslJaasClientSectionName"),
+
+	SERVER_TYPE("serverType"),
+
+	TLS_CERT_FILE("tlsCertFile"),
+
+	TLS_KEY_FILE("tlsKeyFile"),
+
+	TOKEN("token");
 
 	private static final Map<String, String> lowerCaseToCamelCase = new HashMap<>();
 
 	static {
-		// Athenz
-		lowerCaseToCamelCase.put("tenantdomain", "tenantDomain");
-		lowerCaseToCamelCase.put("tenantservice", "tenantService");
-		lowerCaseToCamelCase.put("providerdomain", "providerDomain");
-		lowerCaseToCamelCase.put("privatekey", "privateKey");
-		lowerCaseToCamelCase.put("privatekeypath", "privateKeyPath");
-		lowerCaseToCamelCase.put("keyid", "keyId");
-		lowerCaseToCamelCase.put("autoprefetchenabled", "autoPrefetchEnabled");
-		lowerCaseToCamelCase.put("athenzconfpath", "athenzConfPath");
-		lowerCaseToCamelCase.put("principalheader", "principalHeader");
-		lowerCaseToCamelCase.put("roleheader", "roleHeader");
-		lowerCaseToCamelCase.put("ztsurl", "ztsUrl");
-		// Basic
-		lowerCaseToCamelCase.put("userid", "userId");
-		lowerCaseToCamelCase.put("password", "password");
-		// KeyStoreTls
-		lowerCaseToCamelCase.put("keystoretype", "keyStoreType");
-		lowerCaseToCamelCase.put("keystorepath", "keyStorePath");
-		lowerCaseToCamelCase.put("keystorepassword", "keyStorePassword");
-		// OAuth2
-		lowerCaseToCamelCase.put("type", "type");
-		lowerCaseToCamelCase.put("issuerurl", "issuerUrl");
-		lowerCaseToCamelCase.put("privatekey", "privateKey");
-		lowerCaseToCamelCase.put("audience", "audience");
-		lowerCaseToCamelCase.put("scope", "scope");
-		// Sasl
-		lowerCaseToCamelCase.put("sasljaasclientsectionname", "saslJaasClientSectionName");
-		lowerCaseToCamelCase.put("servertype", "serverType");
-		// Tls
-		lowerCaseToCamelCase.put("tlscertfile", "tlsCertFile");
-		lowerCaseToCamelCase.put("tlskeyfile", "tlsKeyFile");
-		// Token
-		lowerCaseToCamelCase.put("token", "token");
+		for (WellKnownAuthParameters param : values()) {
```

Comment:
Ahhh yes! Big improvement. We can tweak it one step further w/ stream API such as: 
```java
private static final Map<String, String> LOWER_CASE_TO_CAMEL_CASE = Arrays.stream(values())
		.map(WellKnownAuthParameters::getCamelCaseKey)
		.collect(Collectors.toMap(Function.identity(), WellKnownAuthParameters::toCamelCaseKey));
```
NOTE: This would require renaming the existing static method to `toCamelCaseKey` as you would need a getter to get the camel case key. 

I always see how far I can get w/ the streams API and sometimes it turns out less readable etc.. I'm good w/ the code as-is also.

Reply:
That removes the ugly `static {...}` block completely, very nice!